addons/point_of_sale/models/pos_order.py

```diff
@@ -66,10 +66,18 @@ def _get_valid_session(self, order):
                         closed_session.id,
                         order['name'],
                         order['amount_total'])
+        rescue_session = PosSession.search([
+            ('name', '=like', '%(RESCUE FOR %(session)s)' % {'session': closed_session.name}),
```

Comment:
I reviewed the patch before that last change, and I think the previous version was fine with `'like', '(RESCUE FOR %(session)s)' % {'session': closed_session.name})`. The use of `=like` was an attempt to be *even more restrictive*, but there is virtually no risk of collision here. It will also not trigger the bug with the Python format string.

I would like to keep the name of the session in the criterion, because it serves as an implicit selector for the correct POS config_id (which the master PR has). Alternatively, if we only search for "(RESCUE FOR" we need to add a criterion on the config_id.